packages/api/endpoints/providers.js

```diff
@@ -103,14 +106,22 @@ function put(event, cb) {
  * @param {Function} cb - aws lambda callback function
  * @returns {Promise<Object>} returns delete response
  */
-function del(event, cb) {
-  const id = event.pathParameters.id;
+async function del(event, cb) {
   const providerModel = new models.Provider();
 
-  return providerModel.get({ id })
-    .then(() => providerModel.delete({ id }))
-    .then(() => cb(null, { message: 'Record deleted' }))
-    .catch(cb);
+  try {
+    await providerModel.delete(event.pathParameters.id);
+  }
+  catch (err) {
+    if (err instanceof AssociatedRulesError) {
```

Comment:
So the only point of this constructor is so we can distinguish between associated rule errors and others?